src/service/resources-impl.js

```diff
@@ -387,15 +391,42 @@ export class Resources {
     this.resources_.push(resource);
 
     if (this.isRuntimeOn_) {
-      // Try to immediately build element, it may already be ready.
-      resource.build(this.forceBuild_);
-      this.schedulePass();
+      if (this.documentReady_) {
+        // Try to immediately build element, it may already be ready.
+        resource.build(this.forceBuild_);
+        this.schedulePass();
+      } else {
+        // Track open elements
+        this.openElements_.push(element);
+      }
```

Comment:
Also, don't forget to clean this array on resource removal - just as we did with other queues.